packages/product-json-to-csv/src/cli.js

```diff
@@ -0,0 +1,239 @@
+    if (fs.existsSync(arg)) {
+      if (arg.match(/\.csv$/i)) return fs.createReadStream(String(arg))
+
+      throw new Error('Invalid file format. Must be CSV file')
+    }
+    throw new Error('File cannot be reached or does not exist')
+  })
+  .option('input', {
+    alias: 'i',
+    default: 'stdin',
+    describe: 'Path from which to read product chunks.',
+  })
+  .coerce('input', arg => {
+    if (arg === 'stdin') return process.stdin
+
+    if (fs.existsSync(arg)) {
+      if (arg.match(/\.json$/i)) return fs.createReadStream(String(arg))
+
+      throw new Error('Invalid input file format. Must be JSON file')
+    }
+    throw new Error('Input file cannot be reached or does not exist')
+  })
+  .option('output', {
+    alias: 'o',
+    default: 'stdout',
+    describe: 'Path to output',
+    type: 'string',
+  })
+  .coerce('output', arg => {
+    if (arg !== 'stdout') return fs.createWriteStream(String(arg))
+
+    return process.stdout
+  })
+  .option('categoryBy', {
+    choices: ['name', 'key', 'externalId', 'namedPath'],
+    default: 'name',
+    describe:
+      'Define which identifier should be used for the categories column.',
+  })
+  .option('categoryOrderHintBy', {
+    choices: ['name', 'key', 'externalId', 'namedPath'],
+    default: 'name',
+    describe: 'Used CSV delimiter.',
+  })
+  .option('fillAllRows', {
+    describe:
+      'Define which identifier should be used for the categoryOrderHints column.',
+    type: 'boolean',
+  })
+  .option('language', {
+    alias: 'l',
+    default: 'en',
+    describe: 'Language used for localised attributes such as category names.',
+    type: 'string',
+  })
+  .option('delimiter', {
+    alias: 'd',
+    default: ',',
+    describe: 'Used CSV delimiter.',
+  })
+  .option('multiValueDelimiter', {
+    alias: 'm',
+    default: ';',
+    describe: 'Used CSV delimiter in multiValue fields.',
+  })
+  .option('productSeparator', {
+    alias: 'ps',
+    default: '\n',
+    describe: 'Marker separating product chunks',
+    type: 'string',
+  })
+  .option('encoding', {
+    alias: 'e',
+    default: 'utf8',
+    describe: 'Encoding used when saving data to output file',
+    type: 'string',
+  })
+  .option('logLevel', {
+    default: 'info',
+    describe: 'Logging level: error, warn, info or debug',
+    type: 'string',
+  })
+  .option('prettyLogs ', {
+    describe: 'Pretty print logs to the terminal',
+    type: 'boolean',
+  })
+  .option('logFile', {
+    default: 'product-json-to-csv.log',
+    describe: 'Path to file where logs should be saved',
+    type: 'string',
+  }).argv
+
+// instantiate logger
+const loggerConfig = {
+  level: args.logLevel,
+  prettyPrint: args.prettyLogs,
+}
+const logger = pino(loggerConfig)
+
+// print errors to stderr if we use stdout for data output
+// if we save data to output file errors are already logged by pino
+const logError = error => {
+  const errorFormatter = new PrettyError()
+
+  if (logger.level === 'debug')
```

Comment:
Should we read the `logLevel` from the `args` like [above](https://github.com/commercetools/nodejs/pull/418/files#diff-8ed3566bd92ec3973c8d29d39d2e0ab3R145)?

Reply:
I tried to avoid using the `args` directly in function definitions. I dunno which would be better

Reply:
I'd just treat one as the single source of truth. In this case, I guess the `args` are better suited for that since they are exactly what the user inputs.